benchmarks/discrete_trajectory.cpp

```diff
@@ -0,0 +1,166 @@
+
+// .\Release\x64\benchmarks.exe --benchmark_filter=DiscreteTrajectory
+
+#include "physics/discrete_trajectory.hpp"
+
+#include "base/not_null.hpp"
+#include "benchmark/benchmark.h"
+#include "geometry/frame.hpp"
+#include "ksp_plugin/frames.hpp"
+
+namespace principia {
+namespace physics {
+
+using base::not_null;
+using geometry::Frame;
+using geometry::Handedness;
+using geometry::Inertial;
+using geometry::Instant;
+using geometry::Velocity;
+using ksp_plugin::World;
+using quantities::si::Second;
+
+namespace {
+
+// Creates a trajectory with the given number of steps.
+std::unique_ptr<DiscreteTrajectory<World>> CreateTrajectory(int const steps) {
+  auto trajectory = std::make_unique<DiscreteTrajectory<World>>();
+  DegreesOfFreedom<World> const d(World::origin, Velocity<World>());
+  Instant t;
+  for (int i = 0; i < steps; i++, t += Second) {
+    trajectory->Append(t, d);
+  }
+  return trajectory;
+}
+
+// Forks |parent| at a position |pos| of the way through.
+// |parent| should be nonempty.
+// |pos| should be in [0, 1].
+not_null<DiscreteTrajectory<World>*> ForkAt(DiscreteTrajectory<World>& parent,
+                                            float const pos) {
+  CHECK(!parent.Empty());
+  CHECK_GE(pos, 0);
+  CHECK_LE(pos, 1);
+  Instant const desired_fork_time =
+      parent.t_min() + (parent.t_max() - parent.t_min()) * pos;
+  auto const fork_it = parent.LowerBound(desired_fork_time);
+  return parent.NewForkWithCopy(fork_it->time);
+}
+
+}  // namespace
+
+void BM_DiscreteTrajectoryFront(benchmark::State& state) {
+  std::unique_ptr<DiscreteTrajectory<World>> trajectory = CreateTrajectory(4);
```

Comment:
const the pointers that are const (`not_null<...> const` or `auto const`, everywhere in this file).